Server/VikingVault.Services/TimedPaymentService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using VikingVault.DataAccess;
+using VikingVault.DataAccess.Models;
+using VikingVault.Services.Abstractions;
+
+namespace VikingVault.Services
+{
+    public class TimedPaymentService : IHostedService
+    {
+        private Timer _timer;
+        public IServiceProvider Services { get; }
+        public int PaymentCheckIntervalInSeconds = 5 * 60;
+
+        public TimedPaymentService(IServiceProvider services)
+        {
+            Services = services;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+             _timer = new Timer(DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(PaymentCheckIntervalInSeconds));
+             return Task.CompletedTask;         
+        }
+
+        private void DoWork(object state)
+        {
+            using (var scope = Services.CreateScope())
+            {
+                var scopedProcessingService =
+                    scope.ServiceProvider
+                        .GetRequiredService<IScopedProcessingService>();
+
+                scopedProcessingService.PayCompanies();
+            }
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
```

Comment:
It doesn't seem to me like either async method `StopAsync`, `StartAsync` are doing any asynchronous work

Reply:
Yes, well these are overriden functions from IHostedService, so there is no way I can rename them

Reply:
Moreover, I cannot use another background service interface, I have to create a scope specifically for that type of service